crates/sui-types/src/transaction.rs

```diff
@@ -1868,6 +1868,62 @@ pub struct SenderSignedTransaction {
 }
 
 impl SenderSignedData {
+    /// with cache
+    pub fn verify_message_signature_with_cache(
```

Comment:
i had trouble putting it inside verify_message_signature or stick the lru in VerifyParams bc LRU cannot implement copy. not clean - open to suggestions.